src/core/server/http/http_server.test.ts

```diff
@@ -678,30 +679,18 @@ test('#getBasePathFor() is based on server base path', async () => {
 });
 
 test('#setBasePathFor() cannot be set twice for one request', async () => {
-  const incomingMessage = {
-    url: '/',
-  };
   const kibanaRequestFactory = {
     from() {
-      return KibanaRequest.from(
-        {
-          headers: {},
-          path: '/',
-          raw: {
-            req: incomingMessage,
-          },
-        } as any,
-        undefined
-      );
+      return KibanaRequest.from(httpServerMock.createRawRequest(), undefined);
```

Comment:
Typically, passing `undefined` to a function is redundant, can this just be:

```js
return KibanaRequest.from(httpServerMock.createRawRequest());
```

Reply:
`KibanaRequest.from` declaration marks the second parameter as required
https://github.com/restrry/kibana/blob/5128c9bf3d2fc508d95d27414b912e1d12be5a55/src/core/server/http/router/request.ts#L49
I suspect it was done to force the caller to specify route schema. Although this method is used only by Kibana, so probably we can make it optional. @azasypkin

Reply:
> Although this method is used only by Kibana, so probably we can make it optional.

Yeah, it seems we should do that. And probably mark it as `@internal`?

Reply:
there is no way to make a method as truly `@internal` on `@public` class. instead we can declare `from` method as an internal function, not a class static method